### PR TITLE
Update FTP library to make work 'delete_dir()' function

### PR DESCRIPTION
**Pull request by JavaDeveloperKiev**
This function wasnt working, because first 2 elements of $list array returns links to parent folders, like this:

[0] => '.',
[1] => '..',
here all folders and files inside dir

This was casing errors, and directory cannot be deleted. I have added new condition to check and ignore these links. Now its work.

### CODE REVIEW

**Comment by JavaDeveloperKiev** [comment]
Just FYI, in the latest **stable version** that I downloaded, function **isn't working**, and even dont show any exceptions or errors (!)

File: system/libraries/Ftp.php [hunk 1]
```diff
@@ -486,26 +486,31 @@ public function delete_dir($filepath)
...
 			for ($i = 0, $c = count($list); $i < $c; $i++)
 			{
-				// If we can't delete the item it's probably a directory,
-				// so we'll recursively call delete_dir()
-				if ( ! preg_match('#/\.\.?$#', $list[$i]) && ! @ftp_delete($this->conn_id, $list[$i]))
-				{
-					$this->delete_dir($filepath.$list[$i]);
-				}
+				// First 2 elements of $list array would be '.' and '..', this will case errors
+				if($list[$i] != '.' && $list[$i] != '..'){
```
Comment (narfbg) [inline system/libraries/Ftp.php]: [Styleguide](https://www.codeigniter.com/user_guide/general/styleguide.html).

Also, you're adding unnecessary nesting ... just invert the condition and `continue` inside it.

File: system/libraries/Ftp.php [hunk 2]
```diff
@@ -486,26 +486,31 @@ public function delete_dir($filepath)
...
+					// so we'll recursively call delete_dir()
+					if ( ! preg_match('#/\.\.?$#', $list[$i]) && ! @ftp_delete($this->conn_id, $filepath . $list[$i]))
+					{
+						$this->delete_dir($filepath.$list[$i]);
+					}
+				}	
 			}
-		}
 
-		if (@ftp_rmdir($this->conn_id, $filepath) === FALSE)
```
Comment (narfbg) [inline system/libraries/Ftp.php]: Why is this altered?

**Comment by vlakoff** [comment]
Refs #2997

**Pull request closed without merging**